File: burger-maker-service/src/main/java/com/zenika/tutoriels/observabilitydemo/burgermaker/clients/MeatServiceClient.java

package com.zenika.tutoriels.observabilitydemo.burgermaker.clients;

import com.zenika.tutoriels.observabilitydemo.api.contracts.MeatService;
import org.springframework.cloud.openfeign.FeignClient;

@FeignClient(value = "${services.fridge.name:fridge}", contextId = "meat",
        url = "${services.fridge.url:}", configuration = FridgeErrorClientConfiguration.class)
public interface MeatServiceClient extends MeatService {
}

File: burger-maker-service/src/main/kotlin/com/zenika/tutoriels/observabilitydemo/burgermaker/clients/MeatServiceClient.kt
package com.zenika.tutoriels.observabilitydemo.burgermaker.clients

import org.springframework.web.reactive.function.client.WebClient
import reactor.core.publisher.Mono

class MeatServiceClient(serviceUrl: String, builder: WebClient.Builder) : AbstractFridgeClient(builder, "${serviceUrl}/meats") {
    fun orderMeatByKind(kind: String): Mono<Void> {
        return delete(kind)
    }
}